client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -136,6 +138,95 @@ export const useCommandsArrayWpcom = ( {
 		displayNotice( __( 'SSH/SFTP password reset and copied to clipboard.' ) );
 	};
 
+	const getEdgeCacheStatus = async ( siteId: number ) => {
+		const response = await wpcom.req.get( {
+			path: `/sites/${ siteId }/hosting/edge-cache/active`,
+			apiNamespace: 'wpcom/v2',
+		} );
+
+		return response;
+	};
+
+	const clearEdgeCache = async ( siteId: number ) => {
+		try {
+			const response = await getEdgeCacheStatus( siteId );
+
+			if ( response ) {
+				// If global cache is active, purge the cache
+				await wpcom.req.post( {
+					path: `/sites/${ siteId }/hosting/edge-cache/purge`,
+					apiNamespace: 'wpcom/v2',
+				} );
+				displayNotice( __( 'Succesfully cleared cache.' ) );
+			} else {
+				// If global edge cache is not active, clear WordPress cache
+				dispatch( clearWordPressCache( siteId, 'Cache not active' ) );
+			}
+		} catch ( error ) {
+			displayNotice( __( 'Failed to clear cache.' ), 'is-error' );
+		}
+	};
+
+	// Toggle cache function
+	const toggleEdgeCache = async ( siteId: number, newStatus: boolean ) => {
+		const response = await wpcom.req.post( {
+			path: `/sites/${ siteId }/hosting/edge-cache/active`,
+			apiNamespace: 'wpcom/v2',
+			body: {
+				active: newStatus,
+			},
+		} );
+		return response;
+	};
+
+	const enableEdgeCache = async ( siteId: number ) => {
+		const currentStatus = await getEdgeCacheStatus( siteId );
+
+		// Check if the cache is already active
+		if ( currentStatus ) {
+			// Display a different notice if the cache is already active
+			displayNotice( __( 'Edge cache is already enabled.' ), 'is-success' );
+			return;
+		}
+
+		const { removeNotice: removeLoadingNotice } = displayNotice(
+			__( 'Enabling edge cache…' ),
+			'is-plain',
+			5000
+		);
+		try {
+			await toggleEdgeCache( siteId, true );
+			removeLoadingNotice();
+			displayNotice( __( 'Edge cache enabled.' ) );
+		} catch ( error ) {
+			removeLoadingNotice();
+			displayNotice( __( 'Failed to enable edge cache.' ), 'is-error' );
+		}
+	};
+
+	const disableEdgeCache = async ( siteId: number ) => {
+		const currentStatus = await getEdgeCacheStatus( siteId );
+
+		if ( ! currentStatus ) {
+			displayNotice( __( 'Edge cache is already disabled.' ), 'is-success' );
+			return;
+		}
+
+		const { removeNotice: removeLoadingNotice } = displayNotice(
```

Comment:
The code between enabling and disabling edge cache is a bit repetitive so I am wondering if we should do some abstraction for handling notices...